HabitTracker/Report.cs

```diff
@@ -0,0 +1,149 @@
+@$"
+                SELECT strftime('%w',HabitDate) AS ActiveDay, COUNT(*) AS EntryCount
+                FROM Records
+                GROUP BY ActiveDay
+                ORDER BY EntryCount DESC
+                LIMIT 1;
+            ";
+
+            result = command.ExecuteScalar();
+            DayOfWeek day = result != null ? (DayOfWeek)Convert.ToInt32(result) : 0;
+            Console.WriteLine($"\nThe most active day of the week is: {day}");
+
+            command.CommandText =
+            @$"
+                SELECT Name, HabitDate, HabitId, (JULIANDAY(HabitDate) - JULIANDAY(LAG (HabitDate) OVER (PARTITION BY HabitId ORDER BY HabitDate))) * 24 AS Break
+                FROM Records
+                ORDER BY Break DESC
+                LIMIT 1;
+            ";
+
+            using (var reader = command.ExecuteReader())
+            {
+                while (reader.Read())
+                {
+                    string habitName = reader.GetString(0);
+                    int recordCount = reader.GetInt32(3);
+                    int id = reader.GetInt32(2);
+                    Console.WriteLine($"\nThe longest break was taken for the {habitName} habit with an id of {id} - which was not performed for {recordCount} Hours");
+                }
+            }
+
+
+            Console.WriteLine("\nComletion rate statistics:");
+            Dictionary<int, float> habits = new Dictionary<int, float>();
+            command.CommandText =
+            @$"
+                SELECT Id, Frequency, TimesPerPeriod, StartDate,
+                (JULIANDAY('now') - JULIANDAY(StartDate)) * 24 * TimesPerPeriod /
+                CASE 
+                    WHEN Frequency = 'Daily' THEN 24
+                    WHEN Frequency = 'Weekly' THEN 168
+                    WHEN Frequency = 'Monthly' THEN 720
+                END AS ExpectedNumberOfReps 
+                FROM Habits;
+            ";
+            using (var reader = command.ExecuteReader())
+            {
+                while (reader.Read())
+                {
+                    int id = reader.GetInt32(reader.GetOrdinal("Id"));
+                    float expectedReps = reader.GetFloat(reader.GetOrdinal("ExpectedNumberOfReps"));
+                    habits.Add(id, expectedReps);
+                }
+            }
+            command.CommandText =
+            $@"
+                SELECT Name, HabitId, COUNT(*)
+                FROM Records
+                GROUP BY HabitId;
+            ";
+            using (var reader = command.ExecuteReader())
+            {
+                while (reader.Read())
+                {
+                    string name = reader.GetString(0);
+                    int id = reader.GetInt32(1);
+                    double completionRate = Math.Round((reader.GetInt32(2) / habits[id] * 100), 2);
+                    Console.WriteLine($"The '{name}' habit with an id of {id} completion rate is {completionRate}%.");
+                }
+            }
+
```

Comment:
🟠 **Code Style**

💡 Holy whitespace. Makes your code look a mess and un-professional.

🔧 Remove redundant whitespace. One blank line between methods. No blank lines on cascading `}`